Sofa/Component/Engine/Select/src/sofa/component/engine/select/BaseROI.inl

```diff
@@ -0,0 +1,779 @@
+            for (unsigned j = 0; j < 4; j++)
+            {
+                vertices.push_back(convertToVertex(DataTypes::getCPos(x0[q[(j + 1) % 4]])));
+            }
+
+        }
+        vparams->drawTool()->drawLines(vertices, sizeFactor, color);
+    }
+
+    ///draw tetrahedra in ROI
+    if( d_drawTetrahedra.getValue())
+    {
+        std::vector<type::Vec3> vertices;
+        ReadAccessor< Data<vector<Tetra> > > tetrahedraInROI = d_tetrahedraInROI;
+        for (const auto& t : tetrahedraInROI)
+        {
+            type::Vec3 pv{};
+
+            for (unsigned int j=0 ; j<4 ; j++)
+            {
+                vertices.push_back(convertToVertex(DataTypes::getCPos(x0[t[j]])));
+                vertices.push_back(convertToVertex(DataTypes::getCPos(x0[t[(j + 1) % 4]])));
+            }
+
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[t[0]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[t[2]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[t[1]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[t[3]])));
+        }
+        vparams->drawTool()->drawLines(vertices, sizeFactor, color);
+    }
+
+    ///draw hexahedra in ROI
+    if( d_drawHexahedra.getValue())
+    {
+        std::vector<type::Vec3> vertices;
+        ReadAccessor< Data<vector<Hexa> > > hexahedraInROI = d_hexahedraInROI;
+        for (const auto& h : hexahedraInROI)
+        {
+            for (unsigned int j=0 ; j<8 ; j++)
+            {
+                vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[j]])));
+                vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[(j + 1) % 4]])));
+            }
+
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[0]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[2]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[1]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[3]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[4]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[5]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[6]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[7]])));
+        }
+        vparams->drawTool()->drawLines(vertices, sizeFactor, color);
+    }
+
+    vparams->drawTool()->restoreLastState();
+
+}
+
+template<typename DataTypes, typename Element>
+constexpr auto getCenter(const Element& e, const typename DataTypes::VecCoord& x0) -> typename DataTypes::CPos
+{
+    constexpr auto NumberOfNodes = Element::NumberOfNodes;
+
+    assert(NumberOfNodes > 0);
+
+    typename DataTypes::CPos center{};
+    for (const auto eid : e)
+    {
+        center += DataTypes::getCPos(x0[eid]);
+    }
+
+    center = center / static_cast<typename DataTypes::Real>(NumberOfNodes);
```

Comment:
```suggestion
    center = center / static_cast<Real>(NumberOfNodes);
```

Reply:
This is not in a method (so aliases are not available)